### ISSUE TITLE
training continues with loss nan

### ISSUE THREAD

**Issue by kwhkim** [automl/Auto-PyTorch#51]
I am using `develop` branch,

  and as I train regression model using `AutoNetRegression`,

  I notice that training goes on even when loss reaches nan...

  Since no further training is possible, I think AutoPyTorch should reduce the learning rate or

  reduce the momentum or thing like that and stop outputing loss nan...

  I am using Custom Loss but I do not think this should be any problem...

**Comment by LMZimmer** [comment]
Hey,
thank you for the issue, are you referring to the training loss of the NN?

**Comment by kwhkim** [comment]
Yes.  I switched logging mode to debug and i found sometimes it kept
outputing nan..


2020년 6월 11일 (목) 오후 9:27, LMZimmer <notifications@github.com>님이 작성:

**Comment by LMZimmer** [comment]
This is related to gradient explosion which can occur for bad configurations. The corresponding nets will simply have a bad score and the optimizer can use this information to do its thing.

**Comment by maxmarketit** [comment]
Yes, you are right on that optimizer will take the result `nan` into consideration.

  But I think it would be more efficient if the trainer stop trying to train once it hits `nan` 

  since trainer has minimum time to spend on training.

  Maybe this could be another reason to set minimum budget not too large, I guess.
